react/components/ModalTriggerWrapper.tsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import ModalTrigger from './ModalTrigger'
+import { ModalContextProvider } from './ModalContext'
+
+const ModalTriggerWrapper: React.FC<{}> = props => {
+  return (
+    <ModalContextProvider>
+      <ModalTrigger {...props} />
+    </ModalContextProvider>
+  )
+}
+
+export default ModalTriggerWrapper
```

Comment:
I would delete this component in favor of exporting the `ModalTrigger` component wrapped by a `<ModalContextProvider>`, that's one less component :)

Reply:
It doesn't work if I do it, because the trigger uses the context that it is provided by the `ModalContextProvider`

Reply:
Yes! I know, is does, but that should not break.
If you do something like: 

```tsx
const EnhancedModalTrigger = (props) => (
   <ModalContextProvider>
      <ModalTrigger {...props} />
    </ModalContextProvider>
)

export default EnhancedModalTrigger
```

Your ModalTrigger should work just fine :)